src/main/java/it/gov/pagopa/reminder/consumer/PaymentUpdatesKafkaConsumer.java

```diff
@@ -29,20 +29,15 @@ public void paymentUpdatesKafkaListener(PaymentMessage message) {
 			log.info("Received payment-updates: {}", message);
 			payload = message.toString();
 
-			Reminder oldReminder = reminderService.getPaymentByNoticeNumberAndFiscalCode(message.getNoticeNumber(), 
-					message.getPayeeFiscalCode());
-
-			if(Objects.nonNull(oldReminder)) {
-
-				switch (message.getSource()) {
-				case "payments":
-					oldReminder.setPaidFlag(true);
-					oldReminder.setPaidDate(LocalDateTime.now());
-					break;
-				default:
-					break;
+			if("payments".equalsIgnoreCase(message.getSource())) {
+
+				Reminder reminderToUpdate = reminderService.findById(message.getMessageId());
+				if(reminderToUpdate!=null) {
+					reminderToUpdate.setPaidFlag(true);
+					reminderToUpdate.setPaidDate(LocalDateTime.now());
```

Comment:
I'm wondering about this PaidDate. Do we know when a payment has been effectively paid? AFAIK on payment-updates we publish Avro messages containing this information, right?

Reply:
Fixed.
The payment date is now retrieved from the payment-updates queue. In this regard, the payment-updater project has also been updated.
For the proxy, however, we do not receive this information and therefore it is not set.